test/unit/resources/ssh_key_test.rb

```diff
@@ -0,0 +1,28 @@
+require "inspec/globals"
+require "#{Inspec.src_root}/test/helper"
+require_relative "../../../lib/inspec/resources/ssh_key"
+
+describe Inspec::Resources::SshKey do
+  let(:resource) { MockLoader.new("ubuntu".to_sym).load_resource("ssh_key", "test/fixtures/files/test_ssh_key") }
+
+  it "returns the key type" do
+    _(resource.send("type")).must_match(/rsa/)
+  end
+
+  it "returns the key_length" do
+    _(resource.send("key_length")).must_equal 4096
+  end
+
+  it "returns resource_id" do
+    _(resource.send("resource_id")).must_equal "test/fixtures/files/test_ssh_key"
+  end
+
+  it "returns true if key is private" do
+    _(resource.send("private?")).must_equal true
+  end
+
+  it "returns true if key is public" do
+    _(resource.send("public?")).must_equal true
+  end
```

Comment:
this test seems confusing to me. Why is it returning true in both the cases i.e. private and public for the same ssh key?

Reply:
Here is the reference : https://rubydoc.info/gems/rubysl-openssl/OpenSSL%2FPKey%2FRSA:public%3F